### PR TITLE
Ticket 6 - upload example-real-rule no-dynamic-import (also covers require)

### PR DESCRIPTION
**Pull request by unicop**
- Added the `typescript-eslint` TestRuler, which works behind the scenes with `jest`.
- Generated the `@babel/parser` Node types in case we will need it (although we rely on that who is using our plugin is using the `@typescript-eslint/parser`, so it is probably redundant.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,14 +17,23 @@
...
   },
   "devDependencies": {
+    "@babel/types": "^7.17.12",
     "@types/jest": "^27.5.1",
+    "@types/semver": "^7.3.9",
+    "@typescript-eslint/parser": "^5.25.0",
+    "babel-eslint": "^10.1.0",
     "eslint": "^2 || ^3 || ^4 || ^5 || ^6 || ^7.2.0 || ^8",
     "jest": "^28.1.0",
+    "semver": "^7.3.7",
```
Comment (neriyarden) [inline package.json:36]: I didn't understand exactly what this package does?

File: src/types/babel-ast-nodes.ts [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    UpdateExpression: 'UpdateExpression',
+    V8IntrinsicIdentifier: 'V8IntrinsicIdentifier',
+    VariableDeclaration: 'VariableDeclaration',
+    VariableDeclarator: 'VariableDeclarator',
+    Variance: 'Variance',
+    VoidTypeAnnotation: 'VoidTypeAnnotation',
+    WhileStatement: 'WhileStatement',
+    WithStatement: 'WithStatement',
+    YieldExpression: 'YieldExpression'
+}
```
Comment (neriyarden) [inline src/types/babel-ast-nodes.ts:259]: cool 🙂

**Review by neriyarden — APPROVED** [review]
💪🏻

File: package.json (hunk 1 above)
Comment (unicop) [reply package.json:36]: `semver` is a lib that can verify lib versions, for scripts and so own, I think it is redundant here

**Pull request merged**